WordPress/gutenberg
# Add Back to top block

## What?

Updated October 11.

Adds a basic block with a paragraph containing a link that takes you to the top of the browser window.
Fixes https://github.com/WordPress/gutenberg/issues/50433

## Why?


From the issue:
"The popularity of responsive web design has led to a proliferation of single-column, long-page designs both on mobile and desktop. A consequence of these designs has been the Back to Top button, which is a shortcut that allows users to quickly navigate to the top of the page."

## How?


Adds a new block called Back to top.
- The block has a single block attribute which is the link text.
- It uses RichText with the link feature disabled.
- It sets HTML editing to false in block.json.
- The link in the editor is not functional, and that is intended. Selecting it is only for editing the text.
- The target id, `wp-block-back-to-top`, is added to front of the website so that the block always have a target on the page.
- The JavaScript file is needed to move focus, so that the next tab stop is the first focusable element of the page.

## Considerations

- I have placed this block in the "design" category. This is probably wrong. Perhaps there should be a "utility" category?
- I did not disable the inline image on the RichText. I am concerned that the user will only add an image and no text to use as the link text.

## Testing Instructions





1. Add a back to top block, for example in the site footer.
3. Add a second back to top block: change the text, make it part of it bold, italic etc, add an inline image.
4. Save and confirm that both links display correctly on the front.
5. When activating the link, the view should scroll to the top of the page. 

## Updated Testing Instructions for keyboard
1. To add the block, type in, for example, "back" or "top," and the block can be selected from the shortcut.
2. When the block is added, the focus is at the start of the editable text field. The text can be edited, and the default text can be removed.
3. Pressing shift + tab will move the focus to the block toolbar.
4. Navigate the toolbar with the arrow keys to transform and move the block, style selected text, and open the options menu.
5. On the front, navigate to the link and activate it.
6. The view should scroll to the top of the page.
7. Pressing the tab key should move focus to the first focusable element in the body, though after the WordPress navigation bar.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7422055/149d49c2-87b3-49b6-b0c7-a9ca2a6a7bb5
